task.go

```diff
@@ -108,6 +138,102 @@ type Task struct {
 	FetchURIs         []URI             `json:"fetch"`
 }
 
+// TaskV0FromTask is needed for Go versions < 1.8
+// In go 1.8, TaskV0(Task) would work instead
+func TaskV0FromTask(task *Task) TaskV0 {
+	return TaskV0{
+		TaskCPUs:          task.TaskCPUs,
+		TaskMem:           task.TaskMem,
+		Command:           task.Command,
+		Args:              task.Args,
+		User:              task.User,
+		Environment:       task.Environment,
+		MaskedEnvironment: task.MaskedEnvironment,
+		Image:             task.Image,
+		Volumes:           task.Volumes,
+		Ports:             task.Ports,
+		Status:            task.Status,
+		ID:                task.ID,
+		Name:              task.Name,
+		Network:           task.Network,
+		DNS:               task.DNS,
+		FrameworkID:       task.FrameworkID,
+		AgentID:           task.AgentID,
+		AgentConstraints:  task.AgentConstraints,
+		Hostname:          task.Hostname,
+		Retry:             task.Retry,
+		CallbackURI:       task.CallbackURI,
+		SandboxPath:       task.SandboxPath,
+		AgentIP:           task.AgentIP,
+		AgentPort:         task.AgentPort,
+		ForcePullImage:    task.ForcePullImage,
+		FetchURIs:         task.FetchURIs,
+	}
+}
+
+// TaskV1 defines the API V1 json-structure for the properties of a scheduled task.
+type TaskV1 struct {
```

Comment:
maybe an api package, so the whole server package doesn't have to be imported in places where only the structs are wanted?